app/javascript/mastodon/features/compose/components/character_counter.js

```diff
@@ -1,25 +1,53 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { defineMessages, injectIntl } from 'react-intl';
 import { length } from 'stringz';
 
-export default class CharacterCounter extends React.PureComponent {
+const messages = defineMessages({
+  remaining: { id: 'character_counter.remaining', defaultMessage: 'character remaining' },
+  remaining_plural: { id: 'character_counter.remaining_plural', defaultMessage: 'characters remaining' },
+  exceeded: { id: 'character_counter.exceeded', defaultMessage: 'You have exceeded the character limit by' },
+});
+
+export default @injectIntl
+class CharacterCounter extends React.PureComponent {
 
   static propTypes = {
     text: PropTypes.string.isRequired,
     max: PropTypes.number.isRequired,
+    intl: PropTypes.object.isRequired,
   };
 
-  checkRemainingText (diff) {
+  checkRemainingText(diff, intl) {
```

Comment:
Instead of passing `intl`, I think you should get the prop with something like:

```suggestion
  checkRemainingText(diff) {
    const { intl } = this.props;
```